### PR TITLE
ci: implement CI and CD

### PR DESCRIPTION
**Pull request by teran-joseluis**
# US Related

[US 63](https://tree.taiga.io/project/joseluis-teran-coffeetime/us/63?milestone=393080)

- [Task 111](https://tree.taiga.io/project/joseluis-teran-coffeetime/task/111)

## Description

- #### What did I do?

I had created a `.yml` file for our CI/CD workflow, which has the next jobs; `build`, and `test`, also the **pull request template file** so that each **PR** has a description when it's opened.

- #### How did I do it?
- #### Why did I do it?

## Type of Change

Put an `x` in all the boxes that apply:

- [ ] ✨ New Feature (non-breaking change which adds functionality)
- [ ] 🛠️ Bug Fix (non-breaking change which fixes an issue)
- [ ] ❌ Breaking Change (fix or feature that would cause existing functionality to change)
- [ ] 🧹 Code Refactor
- [ ] ✅ Build Configuration Change
- [ ] 📝 Documentation
- [x] 🗑️ Chore

## Pull Request Checklist

Put an `x` in all the boxes that apply:

- [x] My commit messages are detailed.
- [ ] My code follows the code style of this project.
- [ ] No existing features have been broken without good reason.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests pass.

### CODE REVIEW

File: .github/workflows/check-pull-request.yml [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+        uses: actions/setup-java@v4
+        with:
+          distribution: 'adopt'
+          java-version: '17'
+          cache: 'gradle'
+
+      - name: Build using Gradle Wrapper
+        run: ./gradlew build --no-daemon
+
+  test:
```
Comment (zrock-dev) [inline .github/workflows/check-pull-request.yml:29]: This job requires a checkout step too.

image

**Review by zrock-dev — CHANGES_REQUESTED** [review]
Also add an artifact job

**Comment by teran-joseluis** [comment]
> Also add an artifact job

Done.

File: .github/workflows/check-pull-request.yml (hunk 1 above)
Comment (teran-joseluis) [reply .github/workflows/check-pull-request.yml:29]: Yeah, I forgot to add it. Now it's there.

**Comment by teran-joseluis** [comment]
@zrock-dev This PR should trigger to develop in each push or pull_request.
Can you clarify me that please?

**Pull request closed without merging**